### PR TITLE
make inclusive dtor/postblit available

### PR DESCRIPTION
**Pull request by MartinNowak**
- attribute correct destruction and copying
- important building blocks for manual memory management
- replaces complex template hacks (_getPostblit, _destroyRecursive) and
  thus reduces code size and compile time b/c TypeInfo already uses them

### CODE REVIEW

**Comment by MartinNowak** [comment]
This is needed to finish RefCounted and Unique, please review ASAP.

File: src/clone.c [hunk 1]
```diff
@@ -833,13 +833,13 @@ FuncDeclaration *buildPostBlit(StructDeclaration *sd, Scope *sc)
             ea = new CastExp(loc, ea, Type::tvoid->pointerTo());
 
             Expression *et = getTypeInfo(v->type, sc);
-            et = new DotIdExp(loc, et, Id::postblit);
+            et = new DotIdExp(loc, et, Identifier::idPool("postblit"));
```
Comment (WalterBright) [inline src/clone.c]: huh? This is the reverse of other changes further down.

Comment (MartinNowak) [inline src/clone.c]: I know, but I can only have one `Id::postblit`, and for the sake of consistency I made `Id::postblit` like `Id::ctor` to mean the native mangling.

**Comment by MartinNowak** [comment]
Thx

**Comment by JakobOvrum** [comment]
std.traits.hasElaborateCopyConstructor doesn't seem to be the same as `__traits(hasMember, T, "__xpostblit")`. When is `__xpostblit` available?

**Comment by MartinNowak** [comment]
Yes, when one of the fields has a postblit (called `__fieldPostblit` internally).
Same goes for `__xdtor` and `__fieldDtor`.

**Pull request merged**